cumulus_library/studies/vocab/vocab_icd_builder.py

```diff
@@ -30,49 +31,31 @@ def prepare_queries(self, cursor: object, schema: str, *args, **kwargs):
         """
 
         table_name = "vocab__icd"
-        icd_files = ["ICD10CM_2023AA", "ICD10PCS_2023AA", "ICD9CM_2023AA"]
         path = pathlib.Path(__file__).parent
-
+        icd_files = path.glob("icd/*.bsv")
         headers = ["CUI", "TTY", "CODE", "SAB", "STR"]
-        rows_processed = 0
-        dataset = []
-        created = False
-        for filename in icd_files:
-            with open(f"{path}/{filename}.bsv") as file:
-                # For the first row in the dataset, we want to coerce types from
-                # varchar(len(item)) athena default to to an unrestricted varchar, so
-                # we'll create a table with one row - this make the recast faster, and
-                # lets us set the partition_size a little higher by limiting the
-                # character bloat to keep queries under athena's limit of 262144.
-                reader = csv.reader(file, delimiter="|")
-                if not created:
-                    row = self.clean_row(next(reader), filename)
-                    self.queries.append(
-                        base_templates.get_ctas_query(
-                            schema_name=schema,
-                            table_name=table_name,
-                            dataset=[row],
-                            table_cols=headers,
-                        )
-                    )
-                    created = True
-                for row in reader:
-                    row = self.clean_row(row, filename)
-                    dataset.append(row)
-                    rows_processed += 1
-                    if rows_processed == self.partition_size:
-                        self.queries.append(
-                            base_templates.get_insert_into_query(
-                                table_name=table_name,
-                                table_cols=headers,
-                                dataset=dataset,
-                            )
-                        )
-                        dataset = []
-                        rows_processed = 0
-                if rows_processed > 0:
-                    self.queries.append(
-                        base_templates.get_insert_into_query(
-                            table_name=table_name, table_cols=headers, dataset=dataset
-                        )
-                    )
+        header_types = ["STRING", "STRING", "STRING", "STRING", "STRING"]
+        for file in icd_files:
+            parquet_path = path / f"icd/{file.stem}.parquet"
+            if not parquet_path.is_file():
+                df = pandas.read_csv(file, delimiter="|", names=headers)
+                df.to_parquet(parquet_path)
```

Comment:
so i opted to do this step, rather than converting the files to parquet up front and putting those in source control, to make it easier for third parties to understand what was in these files - kind of making the parquet bit an invisible transport mechanism. Could be talked out of it.

Reply:
I buy it